File: youtube_playlist_randomizer/playlist.py

from youtube_playlist_randomizer.playlistitem import PlayListItem
class Playlist:
    """
    Playlist DTO
    """
    
    def __init__(self, id:str, title:str):
        """
        PLaylist object

        Args:
            id (str): playlist id
            title (str): playlist title
        """
        self.id = id
        self.title = title
        self.items = []
    
    def add_item(self, item: PlayListItem) -> None:
        """
        Adds a playlist item to the playlist

        Args:
            item (PlayListItem): Playlist Object
        """
        self.items.append(item)

File: internal/models/playlist.go
package models

type Playlist struct {
	ID        string
	Title     string
	Items     []PlayListItem
	ItemCount int
}

func NewPlaylist(id, title string) *Playlist {
	return &Playlist{
		ID:    id,
		Title: title,
		Items: make([]PlayListItem, 0),
	}
}

func (p *Playlist) AddItem(item PlayListItem) {
	p.Items = append(p.Items, item)
}
